### ISSUE TITLE
Create template file with all necessary functions to write a test

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#5931]
This file can be stored in `test/k8sT/test-template.go` which contains all skelaton of what a test should have. for example: a before all, a context, etc. Potentially it would be something like

```golang

package k8sTest

import (
	"fmt"

	. "github.com/cilium/cilium/test/ginkgo-ext"
	"github.com/cilium/cilium/test/helpers"

	. "github.com/onsi/gomega"
)

var _ = Describe("K8sTestTemplate", func() {

	var (
		kubectl *helpers.Kubectl
	)

	BeforeAll(func() {
		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)

		_ = kubectl.Apply(helpers.DNSDeployment())

		// Deploy the etcd operator
		err := kubectl.DeployETCDOperator()
		Expect(err).To(BeNil(), "Unable to deploy etcd operator")

		err = kubectl.CiliumInstall(helpers.CiliumDefaultDSPatch, helpers.CiliumConfigMapPatch)
		Expect(err).To(BeNil(), "Cilium cannot be installed")

		ExpectCiliumReady(kubectl)
		By("Waiting for all etcd-operator pods are ready")
		ExpectETCDOperatorReady(kubectl)
	})

	AfterFailed(func() {
		kubectl.CiliumReport(helpers.KubeSystemNamespace,
			"cilium endpoint list")
	})

	JustAfterEach(func() {
		kubectl.ValidateNoErrorsOnLogs(CurrentGinkgoTestDescription().Duration)
	})

	AfterEach(func() {
		ExpectAllPodsTerminated(kubectl)
	})
	It("checks cilium-health status between nodes", func() {
		// WRITE YOUR TEST HERE
	}
```

**Comment by eloycoto** [comment]
Why this is needed? I'm not agree with this at all. If we need a template to write a test we're doing something bad 

Plus if we do something like that it'll be in the list of the test results, etc..

**Comment by aanm** [comment]
@eloycoto the istio test for example doesn't deploy etcdoperator https://github.com/cilium/cilium/blob/master/test/k8sT/istio.go#L89

By having a default where developers could look at it and copy-paste before creating a new test would make it easier to avoid these mistakes.

**Comment by eloycoto** [comment]
The problem will happens TBH, the day that the Istio was added ETCD operator was not in there, so developer will have the same problem.

**Comment by aanm** [comment]
>The problem will happens TBH, the day that the Istio was added ETCD operator was not in there, so developer will have the same problem.

That happens will all code. Kube-dns is also missing so it could have been avoided as well. This will still add value for developers to make it less difficult to write new tests.

**Comment by ianvernon** [comment]
>  the istio test for example doesn't deploy etcdoperator

@aanm fix for that is coming in #5869 FYI

**Comment by aanm** [comment]
@ianvernon it's already fixed
